integrations/unstructured/src/haystack_integrations/components/converters/unstructured/converter.py

```diff
@@ -29,7 +30,7 @@ class UnstructuredFileConverter:
     def __init__(
         self,
         api_url: str = UNSTRUCTURED_HOSTED_API_URL,
-        api_key: Optional[str] = None,
+        api_key: Optional[Secret] = Secret.from_env_var("UNSTRUCTURED_API_KEY"),  # noqa: B008
```

Comment:
```suggestion
        api_key: Optional[Secret] = Secret.from_env_var("UNSTRUCTURED_API_KEY", strict=False),  # noqa: B008
```

To make the Secret optional, you can adopt this approach, as done for example in the [`TransformersSimilarityRanker`](https://github.com/deepset-ai/haystack/blob/327c2d260d19ced93fd50e78b63b1465d089c597/haystack/components/rankers/transformers_similarity.py#L45). When you resolve an env var which is not set, you get None. With `strict=True` (default), you get an error.

You can also follow a different approach, as shown [in the docs](https://docs.haystack.deepset.ai/v2.0/docs/secret-management#example).

Reply:
upps, this API KEY is not mandatory, I missed that - thanks 👍🏽